transformers/modeling_gpt2.py

```diff
@@ -564,6 +564,9 @@ def forward(self, input_ids=None, past=None, attention_mask=None, token_type_ids
 
         return outputs  # (loss), lm_logits, presents, (all hidden_states), (attentions)
 
+    def decode(self, input_ids, **model_kwargs):
```

Comment:
Well the idea here is to reduce code duplication. Why not having this generice decode method in the base class?
Also the generic decode method should be the one handling the sequence generation it-self, not only a duplication of the forward call.

Reply:
Within the context of my changes (which is not valid anymore), I designed it this way so that the sampler raises an exception when one tries to sample with a model that cannot decode (say `BertModel`). Code duplication is not always bad, in this case I saw this as an explicit signal that this class can decode. But I am sure you can do it in an another, maybe better, way.

Reply:
What do you mean by "pulling some weight"?